router/base-http/src/main/scala/io/buoyant/router/http/HeadersLike.scala

```diff
@@ -0,0 +1,19 @@
+package io.buoyant.router.http
+
+/**
+  * Type class for HTTP 1.1/2 headers object
+  * @tparam H Headers object type
+  */
+trait HeadersLike[H] {
+  def toSeq(headers: H): Seq[(String, String)]
+  def contains(headers: H, k: String): Boolean
+  def get(headers: H, k: String): Option[String]
+  def getAll(headers: H, k: String): Seq[String]
+  def add(headers: H, k: String, v: String): Unit
+  def set(headers: H, k: String, v: String): Unit
+  def remove(headers: H, key: String): Seq[String]
+}
+
+object HeadersLike {
+  def apply[H](implicit h: HeadersLike[H]): HeadersLike[H] = h
```

Comment:
I'm not 100% sure but I think you can do `implicitly[HeadersLike[H]]` instead of having this method.